src/RottenOranges.java

```diff
@@ -0,0 +1,70 @@
+
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+// LeetCode 994
+class RottenOranges {
+    // global field for storing the grids
+    private int[][] grid;
+    private int numRows;
+    private int numCols;
+    // global field for BFS
+    private final Queue<Coordinate> rottenOranges = new LinkedList<>();
+    private final int[] dx = {0, 1, 0, -1};
+    private final int[] dy = {1, 0, -1, 0};
```

Comment:
```suggestion
    private static final ImmutableList<Integer> DELTAX = {0, 1, 0, -1};
    private static final ImmutableList<Integer> DELTAY = {1, 0, -1, 0};
```
capitalize const
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names